grails-app/services/org/pih/warehouse/core/LocationIdentifierService.groovy

```diff
@@ -0,0 +1,14 @@
+package org.pih.warehouse.core
+
+class LocationIdentifierService extends IdentifierService {
+
+    @Override
+    String getEntityKey() {
+        return "location"
+    }
+
+    @Override
+    protected Integer countDuplicates(String locationNumber) {
+        return Location.countByLocationNumber(locationNumber)
```

Comment:
Seeing this implementation makes me wonder if the appropriate method name for this is ...
```
Integer countByIdentifier(String identifier);
```
That way it's more precise and we're not implying anything by it's invocation. However, maybe the implication ("I want to know how many duplicates are there") is the point.

Reply:
yeah I wanted to be specific about what we need, which is "how many copies of this id already exist in the database". Though I suppose countByIdentifier implies something similar